### PR TITLE
SonarQ added to android

### PR DESCRIPTION
**Pull request by LucianoCaravajal**
# Android SDK

## Extra Notes

### CODE REVIEW

File: .github/workflows/sonar.yml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+        with:
+          path: ~/.gradle/caches
+          key: ${{ runner.os }}-gradle-${{ hashFiles('**/*.gradle') }}
+          restore-keys: ${{ runner.os }}-gradle
+      - name: Build and analyze
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          SONAR_TOKEN: ${{ secrets.SONARQUBE_TOKEN }}
+          SONAR_HOST_URL: ${{ secrets.SONARQUBE_HOST }}
+        run: cd project; ./gradlew build sonarqube
```
Comment (splitadricejas) [inline .github/workflows/sonar.yml:37]: New line

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -12,4 +12,4 @@ jobs:
       - name: Checkout code
         uses: actions/checkout@v2
       - name: Test with Gradle
-        run: cd project; ./gradlew test
+        run: cd project; ./gradlew test
```
Comment (splitadricejas) [inline .github/workflows/test.yml:15]: New line

**Review by splitadricejas — APPROVED** [review]
Just 2 minor changes

File: .github/workflows/sonar.yml (hunk 1 above)
Comment (Israphel) [reply .github/workflows/sonar.yml:37]: Add the wrapper validation step: https://github.com/marketplace/actions/gradle-wrapper-validation

File: .github/workflows/sonar.yml [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+      - name: Cache SonarQube packages
+        uses: actions/cache@v1
+        with:
+          path: ~/.sonar/cache
+          key: ${{ runner.os }}-sonar
+          restore-keys: ${{ runner.os }}-sonar
+      - name: Cache Gradle packages
+        uses: actions/cache@v1
+        with:
+          path: ~/.gradle/caches
```
Comment (Israphel) [inline .github/workflows/sonar.yml:29]: the setup jdk action can cache gradle packages, you don't need this.

File: .github/workflows/sonar.yml [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+    branches:
+      - '*'
+  pull_request:
+    types: [opened, synchronize, reopened]
+jobs:
+  sonar:
+    name: Sonarqube Scan
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
```
Comment (Israphel) [inline .github/workflows/sonar.yml:13]: new version available

File: .github/workflows/sonar.yml [hunk 5]
```diff
@@ -0,0 +1,37 @@
+name: Build
+on:
+  push:
+    branches:
+      - '*'
+  pull_request:
+    types: [opened, synchronize, reopened]
```
Comment (Israphel) [inline .github/workflows/sonar.yml:7]: you should limit this to PRs and pushes to master and development only.

File: .github/workflows/sonar.yml [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+jobs:
+  sonar:
+    name: Sonarqube Scan
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+      - name: Set up JDK 11
+        uses: actions/setup-java@v1
```
Comment (Israphel) [inline .github/workflows/sonar.yml:17]: new version available

File: .github/workflows/sonar.yml [hunk 7]
```diff
@@ -0,0 +1,37 @@
+name: Build
```
Comment (melinaurruchua) [inline .github/workflows/sonar.yml:1]: it more like Test workflow, or Build and test

Comment (LucianoCaravajal) [reply .github/workflows/sonar.yml]: Nice catch, updated, thanks!

**Review by Israphel — DISMISSED** [review]

**Review by Israphel — APPROVED** [review]

**Review by melinaurruchua — APPROVED** [review]

**Pull request merged**